### PR TITLE
stats: Send API key with usage ping

### PR DESCRIPTION
**Pull request by keur**
Resolves https://github.com/brave/brave-browser/issues/9911

## Submitter Checklist:

- [b] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [x] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

1. Get release build of browser so that API_KEY is set
2. Clear browser local state
3. Launch browser and watch network requests (Wireshark, Charles)
4. Ensure that api_key query parameter is set

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_referrals/browser/brave_referrals_service.h [hunk 1]
```diff
@@ -30,7 +30,8 @@ std::string GetAPIKey();
 
 class BraveReferralsService {
  public:
-  explicit BraveReferralsService(PrefService* pref_service);
+  explicit BraveReferralsService(PrefService* pref_service,
+      std::string platform, std::string api_key);
```
Comment (emerick) [inline components/brave_referrals/browser/brave_referrals_service.h:34]: `platform` and `api_key` should be const references.

File: browser/brave_referrals/brave_referrals_service_factory.cc [hunk 2]
```diff
@@ -0,0 +1,31 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
```
Comment (emerick) [inline browser/brave_referrals/brave_referrals_service_factory.cc:3]: Looks like this copyright header is missing a line?

File: components/brave_referrals/browser/brave_referrals_service.h (hunk 1 above)
Comment (keur) [reply components/brave_referrals/browser/brave_referrals_service.h:34]: I made them `const` and force pushed. Both of those are on the stack in the caller (factory function). Both the values will be elided

Comment (emerick) [reply components/brave_referrals/browser/brave_referrals_service.h:34]: Ohh, I was just talking about the params to this function not the member variables. If you make them `const std::string&` in this function, then we won't pass the entire string on the stack. Not a big deal, but it's usually a good idea.

Comment (keur) [reply components/brave_referrals/browser/brave_referrals_service.h:34]: https://github.com/brave/brave-core/blob/f979e0a9d688abc0bbfbf4fae02b36bf935d4795/browser/brave_referrals/brave_referrals_service_factory.cc#L19-L20

I mean both those values are on the stack so a reference won't be valid. But the params are a function call and those will both be copy-elided- it should be the same overhead as using a reference.

**Comment by keur** [comment]
Note: This passed CI in https://ci.brave.com/job/brave-core-build-pr/job/PR-5721/4/, and the new build error was triggered upstream

**Pull request merged**